common/src/main/java/io/netty/util/internal/NativeLibraryLoader.java

```diff
@@ -43,13 +43,13 @@
     private static final InternalLogger logger = InternalLoggerFactory.getInstance(NativeLibraryLoader.class);
 
     private static final String NATIVE_RESOURCE_HOME = "META-INF/native/";
-    private static final String OSNAME;
+    private static final boolean OSX;
     private static final File WORKDIR;
     private static final boolean DELETE_NATIVE_LIB_AFTER_LOADING;
 
     static {
-        OSNAME = SystemPropertyUtil.get("os.name", "").toLowerCase(Locale.US).replaceAll("[^a-z0-9]+", "");
-
+        String osname = SystemPropertyUtil.get("os.name", "").toLowerCase(Locale.US);
+        OSX = osname.startsWith("macosx") || osname.startsWith("osx");
```

Comment:
Yip understood. My rational is if we need todo macOS specific stuff in the future `PlatformDependent` is the natural place to look, but we may miss it (and duplicate it) if the logic lives in this class. Since `PlatformDependent` is internal and duplication removal was the motivation for this PR it seemed like it warranted a discussion.